src/attr/_make.py

```diff
@@ -2727,11 +2762,15 @@ def make_class(name, attrs, bases=(object,), **attributes_arguments):
         raise TypeError("attrs argument must be a dict or a list.")
 
     post_init = cls_dict.pop("__attrs_post_init__", None)
-    type_ = type(
-        name,
-        bases,
-        {} if post_init is None else {"__attrs_post_init__": post_init},
-    )
+    user_init = cls_dict.pop("__init__", None)
```

Comment:
Why do you remove it and put it back later? That seems like something that could break under certain conditions and seems ineffective?

Reply:
I'm just following the pattern that was already present for `post_init`/`__attrs_post_init__`: https://github.com/indigoviolet/attrs/blob/vi-attr-init/src/attr/_make.py#L2764

It looks like we use the [`body` to construct `type_`](https://github.com/indigoviolet/attrs/blob/vi-attr-init/src/attr/_make.py#L2773), and then we pass the [`cls_dict` object to `_attrs`](https://github.com/indigoviolet/attrs/blob/vi-attr-init/src/attr/_make.py#L2797) -- I presume we can't just leave these methods in `cls_dict`. I haven't dug into this.